### PR TITLE
fix format display bugs occurred in DateFormat

### PR DESCRIPTION
**Pull request by XuHuaiyu**
For DateFormat("2010-01-01 00:00:00", "%H %f"), we should get `00 000000`,
but we get `0 0`.
@shenli @coocood @zimulala @hanfei1991 @tiancaiamao PTAL

### CODE REVIEW

**Comment by hanfei1991** [comment]
LGTM

File: evaluator/builtin_time_test.go [hunk 1]
```diff
@@ -217,13 +220,11 @@ func (s *testEvaluatorSuite) TestDateFormat(c *C) {
...
-	// error
 	ds := types.MakeDatums("0000-01-00 00:00:00.123456",
 		"%b %M %m %c %D %d %e %j %k %h %i %p %r %T %s %f %U %u %V %v %a %W %w %X %x %Y %y %%")
-	_, err := builtinDateFormat(ds, nil)
-	// Some like dayofweek() doesn't support the date format like 2000-00-00 returns 0,
-	// so it returns an error.
-	c.Assert(err, NotNil)
+	v, err := builtinDateFormat(ds, nil)
+	c.Assert(err, IsNil)
+	c.Assert(v, testutil.DatumEquals, types.Datum{})
```
Comment (zimulala) [inline evaluator/builtin_time_test.go]: The result of MySQL is 'Jan January 01 1 0th 00 0 000 0 12 00 AM 12:00:00 AM 00:00:00 00 123456 52 00 52 52 Sat Saturday 6 4294967295 4294967295 0000 00 % '

File: evaluator/builtin_time.go [hunk 2]
```diff
@@ -280,14 +286,16 @@ func builtinDateFormat(args []types.Datum, _ context.Context) (types.Datum, erro
 		d         types.Datum
 	)
 
-	// TODO: Some invalid format like 2000-00-01(the month is 0) will return null.
```
Comment (zimulala) [inline evaluator/builtin_time.go]: If it is resolved, add some tests.

Comment (XuHuaiyu) [inline evaluator/builtin_time.go]: I'll resolve this situation in next PR.

**Comment by coocood** [comment]
LGTM

**Comment by zimulala** [comment]
LGTM

**Pull request merged**